### ISSUE TITLE
cmake Failure: Object Meta.BackgroundActor(), has been already deallocated

### ISSUE THREAD

**Issue by grahamPegNetwork** [pop-os/shell#451]
**Distribution (run `cat /etc/os-release`):**
 Ubuntu 20.04 LTS (Fresh install)
 
 
**Related Application and/or Package Version (run `apt policy pop-shell` or provide the latest commit if building locally):**
Commit: dfc745b3f3f733e182eb72153c8dc4bca2d0ff80
Gnome: 3.36.3
tsc --version
Version 3.8.3
make --version
GNU Make 4.2.1

**Issue/Bug Description:**
I started with a fresh Ubuntu 20.04 installation, installed some basic utilities and cloned the repo. Dependencies were installed following instructions found [here](https://www.linuxuprising.com/2020/05/how-to-install-pop-shell-tiling.html). The documented requirements have been installed to the best of my knowledge, but the make script seems to run into an error and crash. The first error I can see (after registering keyboard shortcuts) is:
```
Object Meta.BackgroundActor (0x562d52f3c830), has been already deallocated — impossible to access it.
```
cmake then throws several errors and a stack trace, restarts Gnome, and hangs after outputting:
```
Registering session with GDM
```

The tray icon does not show up, and it appears that the extension is not running. 
I have logged in and out as well as several full reboots, double and triple checking the documented dependencies and removing/re-running the rebuild script.

I'm a big fan of i3/tiling, and have been using i3-regolith for a year or two, but I'm hoping pop shell can take that spot. I met the prerequisites to the best of my knowledge. I would recommend including disto-specific prerequisite installation commands in the future.

**Console output**

```
UUID is "pop-shell@system76.com"
journalctl -o cat -n 0 -f "$(which gnome-shell)" | grep -v warning
Object Meta.BackgroundActor (0x562d52f3c830), has been already deallocated — impossible to access it. This might be caused by the object having been destroyed from C code using something such as destroy(), dispose(), or remove() vfuncs.
== Stack trace for context 0x562d505f36e0 ==
#0   7fff1bff73b0 b   resource:///org/gnome/gjs/modules/core/overrides/GObject.js:571 (331f330b6c40 @ 25)
#1   562d52f8b618 i   /usr/share/gnome-shell/extensions/desktop-icons@csoriano/desktopGrid.js:209 (2d17ee96b50 @ 85)
#2   562d52f8b590 i   /usr/share/gnome-shell/extensions/desktop-icons@csoriano/desktopGrid.js:148 (2d17ee96970 @ 12)
#3   562d52f8b508 i   resource:///org/gnome/shell/ui/main.js:236 (331f330d5e98 @ 12)
JS ERROR: Error: Argument 'instance' (type interface) may not be null
_init/GObject.Object.prototype.disconnect@resource:///org/gnome/gjs/modules/core/overrides/GObject.js:571:24
_onDestroy@/usr/share/gnome-shell/extensions/desktop-icons@csoriano/desktopGrid.js:209:45
_init/<@/usr/share/gnome-shell/extensions/desktop-icons@csoriano/desktopGrid.js:148:44
_initializeUI/<@resource:///org/gnome/shell/ui/main.js:236:16

JS ERROR: TypeError: actor.get_meta_window(...) is null
_destroyWindowDone@resource:///org/gnome/shell/ui/windowManager.js:1590:32
onStopped@resource:///org/gnome/shell/ui/windowManager.js:1560:39
_makeEaseCallback/<@resource:///org/gnome/shell/ui/environment.js:73:13
_easeActor/<@resource:///org/gnome/shell/ui/environment.js:149:56
_initializeUI/<@resource:///org/gnome/shell/ui/main.js:236:16

JS ERROR: TypeError: actor.get_meta_window(...) is null
_destroyWindowDone@resource:///org/gnome/shell/ui/windowManager.js:1590:32
onStopped@resource:///org/gnome/shell/ui/windowManager.js:1560:39
_makeEaseCallback/<@resource:///org/gnome/shell/ui/environment.js:73:13
_easeActor/<@resource:///org/gnome/shell/ui/environment.js:149:56
_initializeUI/<@resource:///org/gnome/shell/ui/main.js:236:16

JS ERROR: TypeError: actor.get_meta_window(...) is null
_destroyWindowDone@resource:///org/gnome/shell/ui/windowManager.js:1590:32
onStopped@resource:///org/gnome/shell/ui/windowManager.js:1560:39
_makeEaseCallback/<@resource:///org/gnome/shell/ui/environment.js:73:13
_easeActor/<@resource:///org/gnome/shell/ui/environment.js:149:56
_initializeUI/<@resource:///org/gnome/shell/ui/main.js:236:16

Unset XDG_SESSION_ID, getCurrentSessionProxy() called outside a user session. Asking logind directly.
Will monitor session 1
Telepathy is not available, chat integration will be disabled.
Error looking up permission: GDBus.Error:org.freedesktop.portal.Error.NotFound: No entry for geolocation
GNOME Shell started at Wed Jul 08 2020 03:26:50 GMT-0700 (PDT)
Registering session with GDM
```
(Hang until Ctrl+C)
```
^Cmake: *** [Makefile:57: listen] Interrupt
```

**Comment by grahamPegNetwork** [comment]
Update: I discovered I wasn't able to enable extensions when trying something else which led me to this thread:
https://www.reddit.com/r/gnome/comments/6gf5d4/question_i_cant_activate_any_of_my_extensions/
I did have gnome-tweak-tools (is that a prerequisite?) and found that there was a slider (enable extensions?) that then allowed me to enable pop shell.

I believe things are now working (I have the tray icon and tiling is working) but the requirements definitely need some polishing/documentation.

As far as I can tell, it looks something like:
1. Install prerequisites (apt install)
2. Install gnome-tweak-tools (?)
3. Open and find/enable extension toggle button (??)
4. Clone repo
5. Run rebuild.sh, Ctrl+C to close (mine stayed open in terminal)
